File: implementations/insertionsort.go
package main

import (
	"fmt"
	// "time"
)

func insertionSort(arr []int) []int {
	for i := 1; i < len(arr); i++ {
		for j := i; j > 0; j-- {
			if arr[j] < arr[j-1] {
				arr[j], arr[j-1] = arr[j-1], arr[j]
			}
		}
	}
	return arr
}
func main() {
	fmt.Println(insertionSort([]int{}))
}
// func main() {
// 	// time calling quicksort on the array
// 	// randarr := rand.Perm(100)
// 	randarr := []int{9,3,6,2,8,7,3,4}
// 	start := time.Now()
// 	insertionSort(randarr)
// 	elapsed := time.Since(start)
// 	// fmt.Println("Array: ", randarr, "\nSorted Array: ", arr, "\nTime elapsed: ", elapsed)
// 	fmt.Println("Time elapsed: ", elapsed)
// }

File: python_impls/insertionsort.py
arr = [6, 5, 3, 1, 8, 7, 2, 4]

def insertion_sort(arr) -> list:
    for i in range(1, len(arr)):
        for j in range(i, 0, -1):
            if arr[j] < arr[j-1]:
                arr[j], arr[j-1] = arr[j-1], arr[j]
    return arr

print(insertion_sort(arr))